src/Microsoft.Azure.WebJobs.Host/Listeners/FunctionListener.cs

```diff
@@ -114,6 +114,17 @@ public void Dispose()
                         message = $"Listener successfully started for function '{_descriptor.ShortName}' after {attempt} retries.";
                         _trace.Info(message);
                         _logger?.LogInformation(message);
+
+                        if (cancellationToken.IsCancellationRequested)
+                        {
+                            // stop has been called while we were in the process of starting
+                            // so we need to stop this listener
+                            await StopAsync(cancellationToken);
+
+                            message = $"Listener for function '{_descriptor.ShortName}' stopped.";
+                            _trace.Info(message);
+                            _logger?.LogInformation(message);
```

Comment:
Can you move this log into `StopAsync()` itself?

Reply:
We're not doing any start/stop/cancel etc. logging for listener operations currently. If we moved this, we'd want to add logging to all the other methods as well. And I think that would result in a lot of noise in the logs. That's why I kept this on the retry path only. What do you think?

Reply:
We can leave it here, that's okay. It may be nitpicky, but I'd suggest we add something to this message that indicates this is only b/c of a retry and that it's not a general 'listener stopping' message. 

My concern was that it could be confusing see "Listener for function 'abc' stopped" under this one specific condition, yet we don't log it generally. Someone analyzing failures in the future may assume that no other listeners were stopped b/c they don't see that log message.

Reply:
Updated message to "Listener for function 'xxx' stopped. A stop was initiated while starting."